sockeye/beam_search.py

```diff
@@ -70,19 +71,32 @@ def encode_and_initialize(self, inputs: pt.Tensor, valid_length: Optional[pt.Ten
     def decode_step(self,
                     step_input: pt.Tensor,
                     states: List,
-                    vocab_slice_ids: Optional[pt.Tensor] = None):
+                    vocab_slice_ids: Optional[pt.Tensor] = None,
+                    target_prefix_factor: Optional[pt.Tensor] = None):
         logits, states, target_factor_outputs = self._model.decode_step(step_input, states, vocab_slice_ids)
         if not self._skip_softmax:
             logits = pt.log_softmax(logits, dim=-1)
-        scores = -logits  # shape: (batch, output_vocab_size/len(vocab_slice_ids))
+        scores = -logits  # shape: (batch*beam, output_vocab_size/len(vocab_slice_ids))
 
         target_factors = None  # type: Optional[pt.Tensor]
         if target_factor_outputs:
+            target_prefix_factor_masks = None  # type: Optional[pt.Tensor]
+            factor_vocab_size = target_factor_outputs[0].size(-1)  # type: ignore
+            if target_prefix_factor is not None:
+                batch_beam, _ = scores.size()
```

Comment:
I wonder if it would make sense to expand `target_prefix_factor` to batch*beam already in GreedySearch/BeamSearch where you know the batch/beam size and don't need to calculate it. What do you think?